### PR TITLE
use prefix for iteration

### PR DESCRIPTION
**Pull request by diptanu**
* Use pre-fetch while iterating (Before - 3.25 seconds while working with a database of 100k rows, after: 18 milliseconds).
* Remove scanning the whole database for getting total count
* Create a reverse index for total tasks, content, etc.

### CODE REVIEW

File: src/state/store/state_machine_objects.rs [hunk 1]
```diff
@@ -865,7 +846,63 @@ impl IndexifyState {
...
+            .get_cf(StateMachineColumns::TaskAnalytics.cf(db), key.clone())
+            .map_err(|e| StateMachineError::DatabaseError(e.to_string()))?;
+        let mut task_analytics: TaskAnalytics = task_analytics
+            .map(|db_vec| {
+                JsonEncoder::decode(&db_vec)
+                    .map_err(|e| StateMachineError::DatabaseError(e.to_string()))
+            })
+            .unwrap_or_else(|| Ok(TaskAnalytics::default()))?;
+        if prev_task.is_none() {
+            task_analytics.pending();
```
Comment (maxkozlovsky) [inline src/state/store/state_machine_objects.rs:882]: increments pending twice, None is handled in match as well

File: src/state/store/mod.rs [hunk 2]
```diff
@@ -755,10 +760,44 @@ impl StateMachineStore {
...
+            read_options,
+            mode,
+        );
+
+        let mut extraction_graph_analytics = ExtractionGraphAnalytics::default();
+
+        for kv in itr {
+            let (key, value) = kv.map_err(|e| anyhow::anyhow!("Failed to read db: {}", e))?;
+            let key = String::from_utf8(key.to_vec())?;
+            let extraction_policy_name = key.strip_prefix(&prefix);
```
Comment (maxkozlovsky) [inline src/state/store/mod.rs:789]: the key is extraction_policy_id, however it is used as name here

File: src/state/store/state_machine_objects.rs (hunk 1 above)
Comment (diptanu) [reply src/state/store/state_machine_objects.rs:882]: Oh that explains why pending count wasn't going down lol. Thanks :)

File: src/state/store/mod.rs (hunk 2 above)
Comment (diptanu) [reply src/state/store/mod.rs:789]: I was looking into it. I don't think we are setting extraction_policy_id anywhere really.

**Pull request merged**